packages/block-editor/src/store/private-selectors.js

```diff
@@ -502,13 +502,23 @@ export const getParentSectionBlock = ( state, clientId ) => {
  * @return {boolean} Whether the block is a content locking parent.
  */
 export function isSectionBlock( state, clientId ) {
+	const blockName = getBlockName( state, clientId );
+	if (
+		blockName === 'core/block' ||
+		getTemplateLock( state, clientId ) === 'contentOnly'
+	) {
+		return true;
+	}
+
+	// Template parts become sections in navigation mode.
+	const _isNavigationMode = isNavigationMode( state );
+	if ( _isNavigationMode && blockName === 'core/template-part' ) {
```

Comment:
Why do template parts and pattern have different behaviors? Should they be "sections" always?

Reply:
Yeah, I'd love to align this more. It mostly comes down to the way synced patterns and blocks with contentOnly template lock are locked from being edited all the time.

Template Parts are still fully editable when editing a template, so they only act as these  `contentOnly` sections in write mode.

This could be aligned more, but I think there's also a UX part to this.

Reply:
Speaking only about template parts, the behaviour in this PR seems intuitive to me in that it makes template parts consistent with current "sections" on trunk: content blocks are editable in write mode but nothing else. On trunk the entire template part is inert.

As for synced patterns, which behaviour _should_ they share? 

I'm still catching up, so is it proposed that the content blocks of synced patterns in write mode should be editable?

I see how the two are related, but I see synced patterns as more discrete objects that require deliberate editing. 

When editing a template, template parts at least in my brain "belong" to the template and so it's logical that their content parts should be editable.

Of course, I could not be making sense at all. 😇